src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -79,4 +81,20 @@ public void storeMessage(Message message) {
 
     return messages;
   }
+
+  /**
+   *Gets list of all users
+   *
+   *@return a Set of users, no order to this list, empty list if there are no users that have posted
+   */
+
+  public Set<String> getUsers(){
+    Set<String> users = new HashSet<>();
+    Query query = new Query("Message");
+    PreparedQuery results = datastore.prepare(query);
+    for(Entity entity : results.asIterable()) {
```

Comment:
Nit: `for (`: https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace, rule 1

(I know that these style issues are from the code on the codeustudents.com documentation; I have sent the team a fixed code snippet.)

You can also install `clang-format` if manually fixing style issues is annoying; then run: `clang-format -i Datastore.java`. There is also probably a plugin for your editor to run a formatter on save.